.githooks/pre-push.bash

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Check if we actually have commits to push
+commits=$(git log @{u}..)
+if [ -z "$commits" ]; then
+    exit 0
+fi
+
+if ! command -v golangci-lint &> /dev/null; then
+    echo "installing golangci-lint..."
+    go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+fi
+
+make lint
+
+make vet
+
+make test-race
```

Comment:
Is it ok to combine this into `make lint vet test-race`?

Reply:
@janos didn't tried, I will test if make will exit when first commands exits with error